src/Symfony/Component/Validator/Constraints/ImageValidator.php

```diff
@@ -43,6 +44,22 @@ public function validate($value, Constraint $constraint)
             return;
         }
 
+        if ($constraint->strict) {
+            if (!function_exists('imagecreatefromstring')) {
+                throw new RuntimeException('Strict image validation requires installed and enabled GD extension');
```

Comment:
same logis uses in email validator: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Validator/Constraints/EmailValidator.php#L57